packages/spruce-skill/server/events/big-search.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+const { eventError } = require('../lib/errorHandler')
+
+const {
+	IBigSearchCtx,
+	IBigSearchSection,
+	IBigSearchResult
+} = require('../types')
+
+module.exports = async (ctx: IBigSearchCtx, next) => {
+	try {
+		console.log('****big-search', ctx.auth.Organization.name)
+
+		const {
+			auth: { Organization: organization, Location: location },
+			event: {
+				payload: { limit, offset, search, testing }
+			}
+		} = ctx
+
+		// each section
+		const sections: IBigSearchSection[] = []
+
+		// do whatever you want with these
+		console.log('ignoring', search, testing)
+
+		// Here is how you could search the core using any rules
+		const { count, rows } = await ctx.db.models.User.findAndCountAll({
+			// where: {
+			// 	firstName: 'Foo',
+			// 	lastName: 'Bar'
+			// }
+			limit,
+			offset
+		})
+
+		const section1: IBigSearchSection = {
+			title: 'Core Search Results Example',
+			section: 'internal',
+			totalCount: count,
+			results: rows.map(
+				(user: Object): IBigSearchResult => ({
+					id: user.id,
+					title: `${user.firstName} ${user.lastName}`,
+					subtitle: ``,
+					action: {
+						type: 'coreRedirect',
+						page: location ? 'profile_user_location' : 'profile_user_org',
+						routeParams: {
+							userId: user.id,
+							organizationId: organization.id,
+							locationId: location && location.id
+						}
+					}
+				})
+			)
+		}
+
+		sections.push(section1)
+
+		// or you can search any source you want and mark them as needing to be imported
+		const dummyResults: IBigSearchResult[] = []
```

Comment:
Probably should improve this to demonstrate pagination, search.

Reply:
We have paging in the query above (lines 33/34), just not in this dummy section. Given i don't have an actual data source to pull from, i'm not quite sure how i'd show paging without baking a lot out.